src/Symfony/Component/HttpKernel/Tests/DependencyInjection/LazyLoadingFragmentHandlerTest.php

```diff
@@ -26,15 +26,25 @@ public function test()
         $requestStack = $this->getMockBuilder('Symfony\Component\HttpFoundation\RequestStack')->getMock();
         $requestStack->expects($this->any())->method('getCurrentRequest')->will($this->returnValue(Request::create('/')));
 
-        $container = $this->getMockBuilder('Symfony\Component\DependencyInjection\ContainerInterface')->getMock();
+        $container = $this->getMockBuilder('Psr\Container\ContainerInterface')->getMock();
+        $container->expects($this->once())->method('has')->with('foo')->willReturn(true);
         $container->expects($this->once())->method('get')->will($this->returnValue($renderer));
 
         $handler = new LazyLoadingFragmentHandler($container, $requestStack, false);
-        $handler->addRendererService('foo', 'foo');
 
         $handler->render('/foo', 'foo');
 
         // second call should not lazy-load anymore (see once() above on the get() method)
         $handler->render('/foo', 'foo');
     }
+
+    /**
+     * @group legacy
+     * @expectedDeprecation The Symfony\Component\HttpKernel\DependencyInjection\LazyLoadingFragmentHandler::addRendererService() method is deprecated since version 3.3 and will be removed in 4.0.
+     */
+    public function testAddRendererServiceIsDeprecated()
+    {
+        $handler = new LazyLoadingFragmentHandler($this->getMockBuilder('Psr\Container\ContainerInterface')->getMock(), $this->getMockBuilder('Symfony\Component\HttpFoundation\RequestStack')->getMock());
+        $handler->addRendererService('foo', 'bar');
```

Comment:
you must keep the existing tests of the class here, to ensure that the 3.2 way of using the class keeps working